app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,13 @@
 class PostsController < ApplicationController
+  before_action :authenticate_user!
+
   def index
     @user = selected_user
     @posts = @user.posts
   end
```

Comment:
- [ ] To complete the configuration of the used gem and as required for the project, kindly ensure that you use **CanCanCan** for `Post` authorization in `Post Controller` like in the example below:
```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```